src/Database/LSMTree/Internal/Merge.hs

```diff
@@ -85,62 +92,103 @@ data MergeState =
 -- and inserts act like mupserts, so they need to be merged monoidally using
 -- 'resolveValue'.
 --
-data MergeType = MergeMidLevel | MergeLastLevel | MergeUnion
+data MergeType = MergeTypeMidLevel | MergeTypeLastLevel | MergeTypeUnion
   deriving stock (Eq, Show)
 
 instance NFData MergeType where
+  rnf MergeTypeMidLevel  = ()
+  rnf MergeTypeLastLevel = ()
+  rnf MergeTypeUnion     = ()
+
+class IsMergeType t where
+  -- TODO: have @isLastLevel :: t -> Bool@ and @isUnion :: t -> Bool@ instead,
+  -- like in the the prototype?
+  toMergeType :: t -> MergeType
```

Comment:
This would probably also be good to do in this PR. Then, instead of storing the `MergeType`, `Merge` would contain `mergeIsLastLevel` and `mergeIsUnion` fields. Ideally not just Boolean, though.

Reply:
(done)